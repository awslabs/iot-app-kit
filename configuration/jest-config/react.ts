export default {
  setupFilesAfterEnv: ['jest-extended/all'],
  maxWorkers: '50%',
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  collectCoverage: true,
  testEnvironment: 'jsdom',
  transform: {
    '.*\\.(tsx?|jsx?)$': [
      '@swc/jest',
      {
        jsc: {
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
        },
      },
    ],
  },
  transformIgnorePatterns: [],
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
  },
};
