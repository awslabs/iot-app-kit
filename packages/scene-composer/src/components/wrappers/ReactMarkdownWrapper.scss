@use '@awsui/design-tokens/index.scss' as awsui;

.markdown-wrapper {
  a {
    color: awsui.$color-text-link-default
  }
  img {
    max-width: 100%;
  }
}
