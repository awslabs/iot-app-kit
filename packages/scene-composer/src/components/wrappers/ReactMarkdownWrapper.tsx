import React, { FC } from 'react';
import ReactMarkdown from 'react-markdown';
import rehypeSanitize from 'rehype-sanitize';

import './ReactMarkdownWrapper.scss';

export interface ReactMarkdownWrapperProps {
  content: string;
  className?: string;
}
export const ReactMarkdownWrapper: FC<ReactMarkdownWrapperProps> = ({ content, className }) => {
  return (
    <ReactMarkdown
      skipHtml
      className={`markdown-wrapper ${className}`}
      linkTarget='_blank'
      rehypePlugins={[rehypeSanitize]}
    >
      {content}
    </ReactMarkdown>
  );
};
