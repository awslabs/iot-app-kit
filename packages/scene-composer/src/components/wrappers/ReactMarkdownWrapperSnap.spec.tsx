import React from 'react';
import { render } from '@testing-library/react';

import { ReactMarkdownWrapper } from './ReactMarkdownWrapper';

describe('ReactMarkdownWrapper', () => {
  it('should render markdown correctly', () => {
    const { container } = render(<ReactMarkdownWrapper content='# header' className='my-class' />);
    expect(container).toMatchSnapshot();
  });
});
