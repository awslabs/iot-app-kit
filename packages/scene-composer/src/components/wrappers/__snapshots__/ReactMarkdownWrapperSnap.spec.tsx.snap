// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMarkdownWrapper should render markdown correctly 1`] = `
<div>
  <div
    class="markdown-wrapper my-class"
  >
    <h1>
      header
    </h1>
  </div>
</div>
`;
